docs/source/examples/full_workflows/plot_public_transit_assignment.py

```diff
@@ -117,38 +127,60 @@
 # %%
 # Mock demand matrix
 # ~~~~~~~~~~~~~~~~~~
-# We'll create a mock demand martix with demand `1` for every zone.
+# We'll create a mock demand matrix with demand `1` for every zone.
 # We'll also need to convert from `zone_id`s to `node_id`s.
```

Comment:
@Jake-Moss  & @djfrancesco, what is this?  Aren't we using zone IDs?  If not, that makes it VERY difficult to use the software

Reply:
Maybe I've worded this comment poorly. AFAIU in aequilibre demand matrices are done centroid to centroid. The transit graph builder takes zones and constructs centroids in them. These are either "od" or "origin" and "destination" node types. The Transit assignment deals with blocking centroid flow by making this distinction (without blocking flow only "od"s are constructed). In the `to_transit_graph` method these centroids are set as the centroids, but because AequilibraE matrices only have a single index between both rows and cols we translate the node ids of the centroids (which are only "od"s regardless of centroid flow) to the corresponding node ids of the "destination"/"origin" node for the assignment

Reply:
The relevant pieces of code are
- `create_od_vertices` https://github.com/Jake-Moss/aequilibrae/blob/8798f47e4f69ae0f9da46a3a4cbbfc096f3d6f92/aequilibrae/transit/transit_graph_builder.py#L503
- `to_transit_graph` https://github.com/Jake-Moss/aequilibrae/blob/8798f47e4f69ae0f9da46a3a4cbbfc096f3d6f92/aequilibrae/transit/transit_graph_builder.py#L1415
- `create_od_node_mapping` https://github.com/Jake-Moss/aequilibrae/blob/8798f47e4f69ae0f9da46a3a4cbbfc096f3d6f92/aequilibrae/transit/transit_graph_builder.py#L544
- Optimal strategies https://github.com/Jake-Moss/aequilibrae/blob/8798f47e4f69ae0f9da46a3a4cbbfc096f3d6f92/aequilibrae/paths/optimal_strategies.py#L33-L38

Reply:
Yes, as @Jake-Moss explained, the demand matrix is expressed in terms of zone IDs, while the assignment method is only dealing with demand between graph vertex IDs. This is why there is the `create_od_node_mapping` method, to get the vertex-based matrix from the zone-based one. This method belongs to the `transit_graph_builder` because this is where we have all the information, of zone IDs and their respective centroids. 
Now in order to block centroid flows, we duplicate the zone centroids into distinct origin and destination nodes. Access connectors start from origin nodes, while egress connectors ends at destination nodes. These origin and destination nodes from a same zone are not directly connected in the graph, which is why this is blocking centroid flows.